### PR TITLE
Enhance dev-tools/kubernetes to support multi node clusters

### PR DESCRIPTION
**Pull request by gsantoro**
## What does this PR do?

Provides support for remote debugging Metricbeat in a multi-node K8s cluster either on Kind or GCP cluster.

## Why is it important?

We want to run multiple Metricbeat instances in normal run mode (oner per worker node), plus having a node that run a Metricbeat instance in debugger mode.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] start Tilt to start a Metricbeat instance with remote debugger enabled on a single node in a multi-node k8s cluster

## How to test this PR locally

1. Run elastic stack locally with `elastic-package stack up`
2. Create a multinode Kind cluster with at least 2 worker nodes
3. Run `kubectl taint nodes <node_name> debugger=yes:NoSchedule` on one of the nodes
4. From the folder `dev-tools/kubernetes` run `tilt up` to deploy Metricbeat in a multi node k8s cluster 
5. Connect from your IDE to the remote debugger
6. Observer N metricbeat instances in Elastic correctly running and keep debugging Metricbeat

## Related issues

-

### CODE REVIEW

File: dev-tools/kubernetes/metricbeat/manifest.debug.multi.yaml [hunk 1]
```diff
@@ -228,6 +227,106 @@ spec:
...
+metadata:
+  name: metricbeat-debugger
+  namespace: kube-system
+  labels:
+    k8s-app: metricbeat
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
```
Comment (tetianakravchenko) [inline dev-tools/kubernetes/metricbeat/manifest.debug.multi.yaml:242]: should be `metricbeat` ?

File: dev-tools/kubernetes/README.md [hunk 2]
```diff
@@ -67,28 +69,16 @@ If you want to use a remote debugger with Visual Studio code, you need to provid
...
-the source code, Tilt will:
-
-1. Automatically compile the source code
-2. Live sync the new binary to the running container
-3. Restart the container to run the new binary
+The behavior of the Tiltfile can be changed by calling the function `beat()` with different parameters:
+- `beat`: `metricbeat` to test Metricbeat, `filebeat` to test Filebeat
+- `mode`: `debug` to start a remote debugger that you can connect to from your IDE with hot reloading enabled, `run` to just run Metricbeat without a debugger but still with hot reloading enabled
+- `arch`: `amd64` to build go binary for amd64 architecture, `arm64` to build go binary for arm64 (aka M1 Apple chip) architecture
+- `k8s_env`: `kind` to run against a Kind cluster with no docker repo, `gcp` to use a docker repo on GCP
```
[the remark this answers is not in the document]
Comment (gsantoro) [reply dev-tools/kubernetes/README.md:76]: Nice catch. I meant `docker registry`

File: dev-tools/kubernetes/README.md [hunk 3]
```diff
@@ -67,28 +69,16 @@ If you want to use a remote debugger with Visual Studio code, you need to provid
...
+- `k8s_cluster`: `single` to use a single node k8s cluster, `multi` to use a k8s with more than 1 node.
```
[the remark this answers is not in the document]
Comment (gsantoro) [reply dev-tools/kubernetes/README.md:77]: k8s_cluster is independent from k8s_env. You can use it on any k8s cluster

**Review by tetianakravchenko — APPROVED** [review]
LGTM!

**Pull request merged**